File: app/src/main/java/com/zionhuang/music/ui/widgets/BottomSheetListener.java
package com.zionhuang.music.ui.widgets;

import android.view.View;

import androidx.annotation.NonNull;

import com.google.android.material.bottomsheet.BottomSheetBehavior;

public interface BottomSheetListener {
    void onStateChanged(@NonNull View bottomSheet,@BottomSheetBehavior.State int newState);
    void onSlide(@NonNull View bottomSheet, float slideOffset);
}

File: app/src/main/java/com/zionhuang/music/ui/widgets/BottomSheetListener.kt
package com.zionhuang.music.ui.widgets

import android.view.View
import com.google.android.material.bottomsheet.BottomSheetBehavior

interface BottomSheetListener {
    fun onStateChanged(bottomSheet: View, @BottomSheetBehavior.State newState: Int)
    fun onSlide(bottomSheet: View, slideOffset: Float)
}